### ISSUE TITLE
typo in ipaclient/tasks/install.yml and length == 0 breaks role

### ISSUE THREAD

**Issue by RobVerduijn** [freeipa/ansible-freeipa#77]
hello.

there is a typo in ipaclient/tasks/install.yml
line 163
```
and (ipaclient_keytab is undefined or ipaclient_keytab|lenth == 0)
```
should be
```
and (ipaclient_keytab is undefined or ipaclient_keytab|length == 0)
```
after replacing that it now fails with :

"msg": "The conditional check 'not result_ipaclient_test_keytab.krb5_keytab_ok and (ipaadmin_password is undefined or ipaadmin_password|length == 0) and (ipaclient_keytab is undefined or ipaclient_keytab|length == 0)' failed. The error was: Unexpected templating type error occurred on ({% if not result_ipaclient_test_keytab.krb5_keytab_ok and (ipaadmin_password is undefined or ipaadmin_password|length == 0) and (ipaclient_keytab is undefined or ipaclient_keytab|length == 0) %} True {% else %} False {% endif %}): object of type 'AnsibleVaultEncryptedUnicode' has no len()

changing lines 162,163
from
```
            (ipaadmin_password is undefined or ipaadmin_password|length == 0)
            and (ipaclient_keytab is undefined or ipaclient_keytab|lenth == 0)
```
to
```
            (ipaadmin_password is undefined or ipaadmin_password == "") and
            (ipaclient_keytab is undefined or ipaclient_keytab == "")
```

fixes that. ( undoing an old commit  #93f9b90)

Rob

**Comment by t-woerner** [comment]
ansible galaxy is complaining on empty string tests. This is the reason why I have replaced them.

**Comment by t-woerner** [comment]
Are you using an encrypted password?

**Comment by RobVerduijn** [comment]
yes, I use inline encrypted vault entries

**Comment by RobVerduijn** [comment]
I guess a check for testing if it is an encrypted string before testing for zero length would do

**Comment by bravosierrasierra** [comment]
typo in second "lenth" instead length 

            (ipaadmin_password is undefined or ipaadmin_password|length == 0)
            and (ipaclient_keytab is undefined or ipaclient_keytab|lenth == 0)

**Comment by RobVerduijn** [comment]
Your post doesn't make sense.
Is this a copy paste typo ?

**Comment by kuwv** [comment]
Typo still exists:
``` (ipaclient_keytab is undefined or ipaclient_keytab|lenth == 0) ```
```
               The error appears to be in '/home/vagrant/.ansible/roles/freeipa.ipaclient/tasks/install.yml': line 159, column 7, but may
               be elsewhere in the file depending on the exact syntax problem.
               
               The offending line appears to be:
               
               
                   - name: Install - Check if one of password and keytab are set
                     ^ here
```

**Comment by RobVerduijn** [comment]
This works again with the latest git version.

**Comment by RobVerduijn** [comment]
this was already fixed in git, and is also working in 1.5.0
